crates/starknet/src/blocks/mod.rs

```diff
@@ -54,6 +55,17 @@ impl StarknetBlocks {
         self.num_to_block.insert(block_number, block);
         self.last_block_hash = Some(hash);
     }
+
+    pub fn block_number_from_block_id(&self, block_id: &BlockId) -> Option<BlockNumber> {
+        match block_id {
+            BlockId::Number(number) => Some(BlockNumber(*number)),
+
+            BlockId::Hash(hash) => self.hash_to_num.get(&Felt::from(*hash)).cloned(),
+
+            BlockId::Tag(BlockTag::Pending) => None,
+            BlockId::Tag(BlockTag::Latest) => Some(BlockNumber(self.hash_to_num.len() as u64)),
```

Comment:
using the length of the collection will not produce always the desired result. Because if the assigned block number doesn't equal the length of the collection the test will fail.
Possible solution:
use the last block hash in the blocks collection to retrieve the latest block number